src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/sampled/compressor_model_sampled.py

```diff
@@ -198,6 +199,13 @@ def evaluate_rate_ps_pd(
         ps_to_evaluate = suction_pressure[indices_to_evaluate] if suction_pressure is not None else []
         pd_to_evaluate = discharge_pressure[indices_to_evaluate] if discharge_pressure is not None else []
 
+        requested_inlet_pressure = (
+            list(ps_to_evaluate) if list(ps_to_evaluate) else [math.nan] * len(list(indices_to_evaluate))
+        )
+        requested_outlet_pressure = (
+            list(pd_to_evaluate) if list(pd_to_evaluate) else [math.nan] * len(list(indices_to_evaluate))
```

Comment:
Are we always consistent in when we add a vector with n `NaN`s and when we just set `null/None`? It is easier to deal with n values, but we might want to reconsider this as it causes a lot of excess "no data" ... just sth to think about